setup.py

```diff
@@ -325,26 +325,28 @@ def read_readme() -> str:
 
 def get_requirements() -> List[str]:
     """Get Python package dependencies from requirements.txt."""
-    with open(get_path("requirements-common.txt")) as f:
-        common = f.read().strip().split("\n")
 
-    if _is_cuda():
-        with open(get_path("requirements-cuda.txt")) as f:
+    def _read_requirements(filename: str) -> List[str]:
+        with open(get_path(filename)) as f:
             requirements = f.read().strip().split("\n")
+        for line in requirements:
+            if line.startswith("-r "):
+                requirements.remove(line)
+                requirements += _read_requirements(line.split()[1])
+        return requirements
```

Comment:
It's best not to mutate the list while iterating:
```suggestion
        resolved_requirements = []
        for line in requirements:
            if line.startswith("-r "):
                resolved_requirements += _read_requirements(line.split()[1])
            else:
                resolved_requirements.append(line)
        return resolved_requirements
```

Reply:
Thanks for the great suggestion! Fixed it.